multichain-aggregator/multichain-aggregator-logic/src/page_token.rs

```diff
@@ -0,0 +1,186 @@
+use alloy_primitives::{Address, B256};
+use base64::Engine;
+use bigdecimal::BigDecimal;
+use chrono::NaiveDateTime;
+use sea_orm::prelude::Decimal;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum PageTokenParsingError {
+    #[error("base64 decode error: {0}")]
+    Base64DecodeError(String),
+    #[error("json decode error: {0}")]
+    JsonDecodeError(String),
+    #[error("invalid format, expected {0} values")]
+    FormatError(u32),
+    #[error("invalid format '{0}' in part #{1}")]
```

Comment:
What do you think about adding some details on at least what type was expected?

like
```
    #[error("invalid format '{0}' in part #{1}; expected `{2}` type")]
    ParsingError(String, u32, String),
```

and `stringify!` in macros